File: src/main/java/de/pcrp/bot/common/EmbedFactory.java
package de.pcrp.bot.common;

import net.dv8tion.jda.api.EmbedBuilder;
import net.dv8tion.jda.api.entities.MessageEmbed;

import java.awt.Color;
import java.time.Instant;

/**
 * Zentrale Stelle für alle Embeds des Bots.
 * Regeln:
 *  - Jedes Embed ist IMMER dunkelorange (0xCC5500).
 *  - Niemals Footer-Texte. Embeds bleiben clean.
 *    Ausnahme: Bot-Neustart-Embed bekommt ein winzig-kleines Java-Logo unten links.
 * Alle Embeds müssen über diese Factory erstellt werden.
 */
public final class EmbedFactory {

    /** Dunkelorange – die einzige erlaubte Embed-Farbe. */
    public static final Color DARK_ORANGE = new Color(0xCC, 0x55, 0x00);

    private EmbedFactory() {}

    /** Leerer Builder in der Standardfarbe. */
    public static EmbedBuilder create() {
        return new EmbedBuilder().setColor(DARK_ORANGE);
    }

    /** Fertiges Embed aus Titel + Beschreibung. */
    public static MessageEmbed build(String title, String description) {
        return create()
                .setTitle(title)
                .setDescription(description)

                .build();
    }

    /**
     * Builder mit winzig-kleinem Icon unten links (Footer-Icon ohne sichtbaren Text).
     * Wird ausschließlich für das Bot-Neustart-Embed verwendet – die einzige Ausnahme
     * von der „kein Footer"-Regel, explizit für Bot-Neustarts gewünscht.
     */
    public static EmbedBuilder createWithFooterIcon(String iconUrl) {
        return create().setFooter("\u200b", iconUrl); // unsichtbares Zeichen als Text
    }
}

File: pcrp-bot/src/PCRP.Bot/Common/EmbedFactory.cs
using Discord;

namespace PCRP.Bot.Common;

/// <summary>
/// Zentrale Stelle für alle Embeds des Bots.
/// Regeln:
///  - Jedes Embed ist IMMER dunkelorange.
///  - Niemals Footer-Texte. Embeds bleiben clean.
/// Alle Embeds müssen über diese Factory erstellt werden.
/// </summary>
public static class EmbedFactory
{
    /// <summary>Dunkelorange – die einzige erlaubte Embed-Farbe.</summary>
    public static readonly Color DarkOrange = new(0xCC, 0x55, 0x00);

    public static EmbedBuilder Create()
        => new EmbedBuilder().WithColor(DarkOrange);

    public static Embed Build(string title, string description)
        => Create()
            .WithTitle(title)
            .WithDescription(description)
            .Build();

    /// <summary>
    /// Erstellt einen EmbedBuilder mit einem winzig-kleinen Icon unten links (Footer-Icon).
    /// Wird ausschließlich für das Bot-Neustart-Embed verwendet – die einzige Ausnahme von
    /// der „kein Footer"-Regel, da der Nutzer dies explizit für Bot-Neustarts gewünscht hat.
    /// Als Footer-Text wird ein unsichtbares Zeichen genutzt, damit das Icon angezeigt wird.
    /// </summary>
    public static EmbedBuilder CreateWithFooterIcon(string iconUrl)
        => Create().WithFooter(new EmbedFooterBuilder()
            .WithIconUrl(iconUrl)
            .WithText("\u200b")); // unsichtbares Zeichen – kein sichtbarer Footer-Text
}
